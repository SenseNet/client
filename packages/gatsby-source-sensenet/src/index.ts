export * from './gatsby-node'
export * from './utils'
