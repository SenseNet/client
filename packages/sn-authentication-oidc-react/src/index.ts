import { UserManager, UserManagerSettings, WebStorageStateStore } from 'oidc-client'
export * from './components/authentication-provider'
export * from './components/oidc-secure'
export * from './use-authentication'
export { WebStorageStateStore, UserManagerSettings, UserManager }
