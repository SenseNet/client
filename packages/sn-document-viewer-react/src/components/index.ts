export * from './comment'
export * from './document-widgets'
export * from './page-widgets'
export * from './ConfirmationDialog'
export * from './DocumentViewer'
export * from './DocumentViewerError'
export * from './DocumentViewerLayout'
export * from './DocumentViewerLoading'
export * from './DocumentViewerRegeneratePreviews'
export * from './LayoutAppBar'
export * from './Page'
export * from './PageList'
export * from './styles'
