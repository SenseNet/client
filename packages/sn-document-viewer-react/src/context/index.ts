export * from './api-settings'
export * from './comment-states'
export * from './comments'
export * from './document-data'
export * from './document-permissions'
export * from './localization-context'
export * from './preview-image-data'
export * from './viewer-settings'
export * from './viewer-state'
