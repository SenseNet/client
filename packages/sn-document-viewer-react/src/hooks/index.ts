export * from './use-comment-state'
export * from './use-comments'
export * from './use-document-data'
export * from './use-document-permissions'
export * from './use-document-viewer-api'
export * from './use-localization'
export * from './use-preview-image'
export * from './use-preview-images'
export * from './use-viewer-settings'
export * from './use-viewer-state'
