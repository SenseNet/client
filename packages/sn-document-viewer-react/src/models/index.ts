export * from './DocumentViewerApiSettings'
export * from './Theming'
export * from './viewer-state'
