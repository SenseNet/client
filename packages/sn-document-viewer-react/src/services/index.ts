export * from './image-utils'
export * from './apply-shape-rotations'
