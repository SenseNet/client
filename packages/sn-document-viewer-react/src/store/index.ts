import { Action, applyMiddleware, compose, createStore, Store } from 'redux'
import { ReduxDiMiddleware } from 'redux-di-middleware'
import { DocumentViewerSettings } from '../models'
import { rootReducer, RootReducerType } from './RootReducer'

const composeEnhancers = (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose

/**
 * gets a configuration object for the Store instance
 * @param {DocumentViewerSettings} settings The Settings object for the document viewer instance
 */
export const getStoreConfig = (settings: DocumentViewerSettings) => {
  const di = new ReduxDiMiddleware(settings.injector)
  di.setInjectable(settings)
  return {
    rootReducer,
    preloadedState: { sensenetDocumentViewer: { documentState: { isLoading: true } } },
    enhancer: composeEnhancers(applyMiddleware(di.getMiddleware())),
  }
}

/**
 * returns a Store object for a Document Viewer instance
 * @param {DocumentViewerSettings} settings The Settings object for the document viewer instance
 */
export const configureStore: (settings: DocumentViewerSettings) => Store<RootReducerType> = settings => {
  const config = getStoreConfig(settings)
  return createStore<RootReducerType, Action, {}, {}>(config.rootReducer, config.preloadedState, config.enhancer)
}

export * from './Comments'
export * from './Document'
export * from './PreviewImages'
export * from './RootReducer'
export * from './Viewer'
