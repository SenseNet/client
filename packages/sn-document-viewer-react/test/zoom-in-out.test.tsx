import IconButton from '@material-ui/core/IconButton/IconButton'
import { mount } from 'enzyme'
import React from 'react'
import { defaultViewerState, ViewerStateContext, ZoomInOutWidget } from '../src'

describe('ZoomInOut component', () => {
  it('should render without crashing', () => {
    const wrapper = mount(<ZoomInOutWidget />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should increase zoomLevel in viewerState on zoom in', () => {
    const updateState = jest.fn()
    const wrapper = mount(
      <ViewerStateContext.Provider
        value={{
          ...defaultViewerState,
          zoomLevel: 0,
          updateState,
        }}>
        <ZoomInOutWidget />
      </ViewerStateContext.Provider>,
    )
    wrapper.find(IconButton).first().simulate('click')
    expect(updateState).toBeCalledWith({
      zoomLevel: 1,
    })
  })

  it('should reduce zoomLevel in viewerState on zoom out', () => {
    const updateState = jest.fn()
    const wrapper = mount(
      <ViewerStateContext.Provider
        value={{
          ...defaultViewerState,
          zoomLevel: 1,
          updateState,
        }}>
        <ZoomInOutWidget />
      </ViewerStateContext.Provider>,
    )
    wrapper.find(IconButton).last().simulate('click')
    expect(updateState).toBeCalledWith({
      zoomLevel: 0,
    })
  })
})
