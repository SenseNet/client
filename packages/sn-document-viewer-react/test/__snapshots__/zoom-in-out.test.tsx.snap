// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZoomInOut component should render without crashing 1`] = `
<ZoomInOutWidget>
  <div
    style={
      Object {
        "display": "inline-block",
      }
    }
  >
    <WithStyles(ForwardRef(IconButton))
      className="makeStyles-iconButton-1"
      color="inherit"
      disabled={false}
      onClick={[Function]}
      title="Zoom in"
    >
      <ForwardRef(IconButton)
        className="makeStyles-iconButton-1"
        classes={
          Object {
            "colorInherit": "MuiIconButton-colorInherit",
            "colorPrimary": "MuiIconButton-colorPrimary",
            "colorSecondary": "MuiIconButton-colorSecondary",
            "disabled": "Mui-disabled",
            "edgeEnd": "MuiIconButton-edgeEnd",
            "edgeStart": "MuiIconButton-edgeStart",
            "label": "MuiIconButton-label",
            "root": "MuiIconButton-root",
            "sizeSmall": "MuiIconButton-sizeSmall",
          }
        }
        color="inherit"
        disabled={false}
        onClick={[Function]}
        title="Zoom in"
      >
        <WithStyles(ForwardRef(ButtonBase))
          centerRipple={true}
          className="MuiIconButton-root makeStyles-iconButton-1 MuiIconButton-colorInherit"
          disabled={false}
          focusRipple={true}
          onClick={[Function]}
          title="Zoom in"
        >
          <ForwardRef(ButtonBase)
            centerRipple={true}
            className="MuiIconButton-root makeStyles-iconButton-1 MuiIconButton-colorInherit"
            classes={
              Object {
                "disabled": "Mui-disabled",
                "focusVisible": "Mui-focusVisible",
                "root": "MuiButtonBase-root",
              }
            }
            disabled={false}
            focusRipple={true}
            onClick={[Function]}
            title="Zoom in"
          >
            <button
              className="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-1 MuiIconButton-colorInherit"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              title="Zoom in"
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <ForwardRef(ZoomInIcon)
                  className="makeStyles-icon-2"
                >
                  <WithStyles(ForwardRef(SvgIcon))
                    className="makeStyles-icon-2"
                  >
                    <ForwardRef(SvgIcon)
                      className="makeStyles-icon-2"
                      classes={
                        Object {
                          "colorAction": "MuiSvgIcon-colorAction",
                          "colorDisabled": "MuiSvgIcon-colorDisabled",
                          "colorError": "MuiSvgIcon-colorError",
                          "colorPrimary": "MuiSvgIcon-colorPrimary",
                          "colorSecondary": "MuiSvgIcon-colorSecondary",
                          "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                          "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                          "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                          "root": "MuiSvgIcon-root",
                        }
                      }
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root makeStyles-icon-2"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                        />
                        <path
                          d="M12 10h-2v2H9v-2H7V9h2V7h1v2h2v1z"
                        />
                      </svg>
                    </ForwardRef(SvgIcon)>
                  </WithStyles(ForwardRef(SvgIcon))>
                </ForwardRef(ZoomInIcon)>
              </span>
              <WithStyles(memo)
                center={true}
              >
                <ForwardRef(TouchRipple)
                  center={true}
                  classes={
                    Object {
                      "child": "MuiTouchRipple-child",
                      "childLeaving": "MuiTouchRipple-childLeaving",
                      "childPulsate": "MuiTouchRipple-childPulsate",
                      "ripple": "MuiTouchRipple-ripple",
                      "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                      "rippleVisible": "MuiTouchRipple-rippleVisible",
                      "root": "MuiTouchRipple-root",
                    }
                  }
                >
                  <span
                    className="MuiTouchRipple-root"
                  >
                    <TransitionGroup
                      childFactory={[Function]}
                      component={null}
                      exit={true}
                    />
                  </span>
                </ForwardRef(TouchRipple)>
              </WithStyles(memo)>
            </button>
          </ForwardRef(ButtonBase)>
        </WithStyles(ForwardRef(ButtonBase))>
      </ForwardRef(IconButton)>
    </WithStyles(ForwardRef(IconButton))>
    <WithStyles(ForwardRef(IconButton))
      className="makeStyles-iconButton-1"
      color="inherit"
      disabled={false}
      onClick={[Function]}
      title="Zoom out"
    >
      <ForwardRef(IconButton)
        className="makeStyles-iconButton-1"
        classes={
          Object {
            "colorInherit": "MuiIconButton-colorInherit",
            "colorPrimary": "MuiIconButton-colorPrimary",
            "colorSecondary": "MuiIconButton-colorSecondary",
            "disabled": "Mui-disabled",
            "edgeEnd": "MuiIconButton-edgeEnd",
            "edgeStart": "MuiIconButton-edgeStart",
            "label": "MuiIconButton-label",
            "root": "MuiIconButton-root",
            "sizeSmall": "MuiIconButton-sizeSmall",
          }
        }
        color="inherit"
        disabled={false}
        onClick={[Function]}
        title="Zoom out"
      >
        <WithStyles(ForwardRef(ButtonBase))
          centerRipple={true}
          className="MuiIconButton-root makeStyles-iconButton-1 MuiIconButton-colorInherit"
          disabled={false}
          focusRipple={true}
          onClick={[Function]}
          title="Zoom out"
        >
          <ForwardRef(ButtonBase)
            centerRipple={true}
            className="MuiIconButton-root makeStyles-iconButton-1 MuiIconButton-colorInherit"
            classes={
              Object {
                "disabled": "Mui-disabled",
                "focusVisible": "Mui-focusVisible",
                "root": "MuiButtonBase-root",
              }
            }
            disabled={false}
            focusRipple={true}
            onClick={[Function]}
            title="Zoom out"
          >
            <button
              className="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-1 MuiIconButton-colorInherit"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              title="Zoom out"
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <ForwardRef(ZoomOutIcon)
                  className="makeStyles-icon-2"
                >
                  <WithStyles(ForwardRef(SvgIcon))
                    className="makeStyles-icon-2"
                  >
                    <ForwardRef(SvgIcon)
                      className="makeStyles-icon-2"
                      classes={
                        Object {
                          "colorAction": "MuiSvgIcon-colorAction",
                          "colorDisabled": "MuiSvgIcon-colorDisabled",
                          "colorError": "MuiSvgIcon-colorError",
                          "colorPrimary": "MuiSvgIcon-colorPrimary",
                          "colorSecondary": "MuiSvgIcon-colorSecondary",
                          "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                          "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                          "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                          "root": "MuiSvgIcon-root",
                        }
                      }
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root makeStyles-icon-2"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14zM7 9h5v1H7z"
                        />
                      </svg>
                    </ForwardRef(SvgIcon)>
                  </WithStyles(ForwardRef(SvgIcon))>
                </ForwardRef(ZoomOutIcon)>
              </span>
              <WithStyles(memo)
                center={true}
              >
                <ForwardRef(TouchRipple)
                  center={true}
                  classes={
                    Object {
                      "child": "MuiTouchRipple-child",
                      "childLeaving": "MuiTouchRipple-childLeaving",
                      "childPulsate": "MuiTouchRipple-childPulsate",
                      "ripple": "MuiTouchRipple-ripple",
                      "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                      "rippleVisible": "MuiTouchRipple-rippleVisible",
                      "root": "MuiTouchRipple-root",
                    }
                  }
                >
                  <span
                    className="MuiTouchRipple-root"
                  >
                    <TransitionGroup
                      childFactory={[Function]}
                      component={null}
                      exit={true}
                    />
                  </span>
                </ForwardRef(TouchRipple)>
              </WithStyles(memo)>
            </button>
          </ForwardRef(ButtonBase)>
        </WithStyles(ForwardRef(ButtonBase))>
      </ForwardRef(IconButton)>
    </WithStyles(ForwardRef(IconButton))>
  </div>
</ZoomInOutWidget>
`;
